### PR TITLE
SymbolItems with @startid and fixed @startid/@endid for slurs within the same measure

### LINKED ISSUE #62: export of ornaments
While turns are exported quite nice, mordents are missing several attributes. There is missing `@staff` and `@layer` and most importantly `@startid` (`@tstamp`).
Long mordents won't get exported at all.  
Notice that stroked turns are exported as inverted turns, while (normal) inverted turns will be ignored. 

Preferably all ornaments should have a `@startid` instead of a `@tstamp`

```xml
<?xml version="1.0" encoding="UTF-16" ?>
<mei xml:id="m-1" meiversion="3.0.0" xmlns="http://www.music-encoding.org/ns/mei" xmlns:xlink="http://www.w3.org/1999/xlink">
<meiHead xml:id="m-2">
    <fileDesc xml:id="m-3">
        <titleStmt xml:id="m-4">
            <title xml:id="m-10"/>
            <respStmt xml:id="m-11">
                <persName xml:id="m-12"/>
            </respStmt>
        </titleStmt>
        <pubStmt xml:id="m-13">
            <availability xml:id="m-14">
                <useRestrict xml:id="m-15">Copyright © </useRestrict>
            </availability>
        </pubStmt>
    </fileDesc>
    <encodingDesc xml:id="m-16">
        <appInfo xml:id="m-17">
            <application xml:id="sibelius" isodate="2017-2-3T16:13:22Z" version="7500">
                <name xml:id="m-19" type="operating-system">Windows 7</name>
            </application>
            <application xml:id="sibmei" type="plugin" version="2.0.2">
                <name xml:id="m-21">Sibelius to MEI Exporter (2.0.2)</name>
            </application>
        </appInfo>
    </encodingDesc>
    <workDesc xml:id="m-5">
        <work xml:id="m-6">
            <titleStmt xml:id="m-7">
                <title xml:id="m-8"/>
                <respStmt xml:id="m-9"/>
            </titleStmt>
        </work>
    </workDesc>
</meiHead>
<music xml:id="m-22">
    <body xml:id="m-23">
        <mdiv xml:id="m-24">
            <score xml:id="m-25">
                <scoreDef xml:id="m-26" lyric.name="Opus Text Std" meter.count="4" meter.unit="4" music.name="Opus Std" page.botmar="12.7mm" page.height="297mm" page.leftmar="12.7mm" page.rightmar="12.7mm" page.topmar="12.7mm" page.width="210mm" ppq="256" text.name="Plantin MT Std">
                    <staffGrp xml:id="m-27">
                        <staffDef xml:id="m-28" clef.line="2" clef.shape="G" key.mode="major" key.sig="0" label="" lines="5" n="1">
                            <!-- keyboard.piano.grand.bright -->
                            <instrDef xml:id="m-30" midi.channel="1" midi.pan="63" midi.volume="100"/>
                        </staffDef>
                    </staffGrp>
                </scoreDef>
                <section xml:id="m-31">
                    <measure xml:id="m-32" label="1" n="1">
                        <staff xml:id="m-33" n="1">
                            <layer xml:id="m-34" n="1">
                                <note xml:id="m-35" dur="4" dur.ges="256p" oct="5" pname="c" pnum="72" stem.dir="down"/>
                                <note xml:id="m-36" dur="4" dur.ges="256p" oct="4" pname="a" pnum="69" stem.dir="up"/>
                                <note xml:id="m-37" dur="4" dur.ges="256p" oct="4" pname="f" pnum="65" stem.dir="up"/>
                                <note xml:id="m-38" dur="4" dur.ges="256p" oct="4" pname="a" pnum="69" stem.dir="up"/>
                            </layer>
                        </staff>
                        <mordent xml:id="m-39" form="inv"/>
                        <mordent xml:id="m-40" form="norm"/>
                        <turn xml:id="m-41" form="norm" layer="1" staff="1" tstamp="3" vo="5.25mm"/>
                        <turn xml:id="m-42" form="inv" layer="1" staff="1" tstamp="4" vo="5.25mm"/>
                    </measure>
                    <measure xml:id="m-43" n="2">
                        <staff xml:id="m-44" n="1">
                            <layer xml:id="m-45" n="1">
                                <note xml:id="m-46" dur="4" dur.ges="256p" oct="5" pname="c" pnum="72" stem.dir="down"/>
                                <note xml:id="m-47" dur="4" dur.ges="256p" oct="4" pname="a" pnum="69" stem.dir="up"/>
                                <note xml:id="m-48" dur="4" dur.ges="256p" oct="4" pname="f" pnum="65" stem.dir="up"/>
                                <note xml:id="m-49" dur="4" dur.ges="256p" oct="4" pname="a" pnum="69" stem.dir="up"/>
                            </layer>
                        </staff>
                        <trill xml:id="m-50" layer="1" staff="1" tstamp="1" vo="5.25mm"/>
                    </measure>
                    <measure xml:id="m-51" n="3">
                        <staff xml:id="m-52" n="1">
                            <layer xml:id="m-53" n="1">
                                <note xml:id="m-54" dur="4" dur.ges="256p" oct="5" pname="c" pnum="72" stem.dir="down"/>
                                <note xml:id="m-55" dur="4" dur.ges="256p" oct="4" pname="b" pnum="71" stem.dir="down"/>
                                <note xml:id="m-56" dur="4" dur.ges="256p" oct="4" pname="a" pnum="69" stem.dir="up"/>
                                <note xml:id="m-57" dur="4" dur.ges="256p" oct="4" pname="d" pnum="62" stem.dir="up"/>
                            </layer>
                        </staff>
                    </measure>
                    <measure xml:id="m-58" n="4" right="end">
                        <staff xml:id="m-59" n="1">
                            <layer xml:id="m-60" n="1">
                                <note xml:id="m-61" dur="4" dur.ges="256p" oct="4" pname="f" pnum="65" stem.dir="up"/>
                                <note xml:id="m-62" dur="4" dur.ges="256p" oct="4" pname="e" pnum="64" stem.dir="up"/>
                                <rest xml:id="m-63" dur="2" dur.ges="512p"/>
                            </layer>
                        </staff>
                    </measure>
                </section>
            </score>
        </mdiv>
    </body>
</music>
</mei>
```
[ornaments.zip](https://github.com/music-encoding/sibmei/files/750823/ornaments.zip)

### PR DESCRIPTION
**Pull request by annplaksin**
This PR provides the export of ornaments with @startid: closes #62 

Additionally the structure of GenerateLayer() is redone to fix the export of @startid and @endid for slurs.
It must be ensured that Line objects are processed after NoteRest obejcts to be able to find the corresponding notes. Usually, if more than one object is at the same position, the NoteRest object is the last one.

### CODE REVIEW

File: src/ExportGenerators.mss [hunk 1]
```diff
@@ -661,69 +662,92 @@ function GenerateLayers (staffnum, measurenum) {
...
-        if (chordsym != null)
-        {
+      // add chord symbols to the measure objects
+      // so that they get added to the measure later in the processing cycle.
+      if (chordsym != null)
+      {
             mlines = Self._property:MeasureObjects;
             mlines.Push(chordsym._id);
             Self._property:MeasureObjects = mlines;
-        }
```
Comment (th-we) [inline src/ExportGenerators.mss]: @annplaksin You changed the indentation here. 👆 :dudu:

**Comment by th-we** [comment]
Works! Wrote a small test and added it as a [pull request to your pull request](https://github.com/maxweberstiftung/sibmei/pull/2).

Minor thing is some indentation – would be cool if you could fix that.

**Comment by th-we** [comment]
Mh, just modified the test to have multiple slurs with identical length in two layers – that case is not handled properly yet. We have to check for NoteRests that have the proper timestamp *and* the proper `VoiceNumber`.

**Comment by th-we** [comment]
I was to rash with my last comment there – works fine, actually!

**Comment by annplaksin** [comment]
@th-we: Is it better now or do you still find any mess?

File: src/ExportGenerators.mss [hunk 2]
```diff
@@ -704,6 +716,16 @@ function GenerateLayers (staffnum, measurenum) {
...
             {
                 mobj = ConvertText(bobj);
+                if (mobj != null)
+                {
+                    //Try to get note at position of bracket and put id
+                    obj = GetNoteObjectAtPosition(bobj);
+
+                    if (obj != null)
+                    {
+                        libmei.AddAttribute(text, 'startid', '#' & obj._id);
```
Comment (th-we) [inline src/ExportGenerators.mss]: @annplaksin `text` should be `mobj`. Can you change this and squash it into commit dd3ba30?

Comment (annplaksin) [reply src/ExportGenerators.mss]: Is done...

**Comment by th-we** [comment]
I modified some minor things in the commits and created a new pull request – could you have a look at it, @annplaksin?

**Pull request closed without merging**